src/components/modal/base/modal.jsx

```diff
@@ -0,0 +1,52 @@
+var classNames = require('classnames');
+var omit = require('lodash.omit');
+var React = require('react');
+var ReactModal = require('react-modal');
+
+require('./modal.scss');
+
+/**
+ * Container for pop up windows (See: registration window)
+ */
+var Modal = React.createClass({
+    type: 'Modal',
+    statics: {
+        setAppElement: ReactModal.setAppElement
+    },
+    propTypes: {
+        className: React.PropTypes.string,
+        overlayClassName: React.PropTypes.string
+    },
+    requestClose: function () {
+        return this.refs.modal.portal.requestClose();
+    },
+    render: function () {
+        var modalClasses = classNames(
+            'modal-content',
+            this.props.className
+        );
+        var overlayClasses = classNames(
+            'modal-overlay',
+            this.props.overlayClassName
+        );
+        return (
+            <ReactModal
+                ref="modal"
```

Comment:
As long as we're refactoring this and testing, we should get rid of this string ref — they are deprecated. Instead make it a function like `ref={function (c){ this.modal = c;}.bind(this)}`, and then up above don't refer to `this.refs.modal`, just refer to `this.modal`